src/components/TopBar/components/WalletButton.tsx

```diff
@@ -89,13 +96,31 @@ const WalletButton: React.FC<WalletButtonProps> = (props) => {
     <>
       <StyledWalletButton>
         {!userAccount ? (
-          <Button onClick={handleUnlockWalletClick} size="sm" text="Unlock Wallet" />
+          <Button
+            onClick={handleUnlockWalletClick}
+            size="sm"
+            text="Unlock Wallet"
+          />
         ) : (
-          <Button onClick={handleWalletClick} size="sm" text="Balances" variant="tertiary" />
+          <Button onClick={handleWalletClick} size="sm" variant="tertiary">
+            <Davatar
+              address={userAccount}
+              size={20}
+              style={{ marginRight: 8 }}
+              generatedAvatarType="jazzicon"
+            />
+            <div>{ensName || shorten(userAccount)}</div>
```

Comment:
for some reason the spacing between the avatar and the address/ens is very close and doesnt look like what you screenshotted, adding more space in between will make it look good!

Reply:
@ethedev I found the issue now, we need to add a new CSS class to handle it 🤗

Reply:
@ethedev this id fixed now by adding `<Spacer />` utility

Reply:
looking good!

Reply:
@ethedev is it good to merge?